Scripts/script-StringReplace.yml

```diff
@@ -0,0 +1,78 @@
+commonfields:
+  id: StringReplace
+  version: -1
+name: StringReplace
+script: |-
+  if ((typeof args.data) !== 'string') {
+      return {
+          Type: entryTypes.error,
+          ContentsFormat: formats.text,
+          Contents: 'Invalid input - please make sure \'data\' is of type string'
+      }
+  }
+
+  var arr = [];
+  if (args.replaceAll === 'true') {
+      arr.push('g');
+  }
+  if (args.caseInsensitive === 'true') {
+      arr.push('i');
+  }
+  if (args.multiLine === 'true') {
+      arr.push('m');
+  }
+
+  var newValue = args.newValue === "''" ? "" : args.newValue;
+  var r = new RegExp(args.regex, arr.join(''));
+  var val = args.data.replace(r, newValue);
+
+  //if no match found - return the original string
+  setContext('StringReplace.Result', val);
+  return {
+      Type: entryTypes.note,
+      Contents: val,
+      ContentsFormat: formats.text
+  };
+type: javascript
+tags:
+- Utilities
+comment: |-
+  Replaces regex match/es in string.
+  Returns the string after replace was preformed.
+enabled: true
+args:
+- name: data
+  required: true
+  default: true
+  description: The string to preform replace on
+- name: regex
+  required: true
+  description: The regex to find matches to be replaced with a new value
+- name: newValue
+  required: true
+  description: The new value to replace the regex match. Pass '' to remove regex match.
+- name: replaceAll
+  auto: PREDEFINED
+  predefined:
+  - "true"
+  - "false"
+  description: The boolean value to indicate if 'g' flag is passed to regex constructor
```

Comment:
I would explain what does g means.. something like, true to replace all, false to replace only the first occurrence.
In most cases users are not programmers.
Same for the other arguments (i and m)